generative_ai/embeddings/document_retrieval_example.py

```diff
@@ -19,27 +19,41 @@
 
 
 def embed_text() -> list[list[float]]:
-    """Embeds texts with a pre-trained, foundational model.
+    """Calculates embeddings for the given text inputs.
 
     Returns:
-        A list of lists containing the embedding vectors for each input text
+        A list of lists containing the embedding vectors for each input text.
     """
 
-    # A list of texts to be embedded.
-    texts = ["banana muffins? ", "banana bread? banana muffins?"]
+    model = TextEmbeddingModel.from_pretrained("text-embedding-004")
+
+    inputs = [
+        # Option1: Use TextEmbeddingInput object as embedding input
+        TextEmbeddingInput(
+            text=(
+                "You'll find some apps on your Home screens, and all your apps in All Apps. "
+                "You can open apps, switch between apps, and find 2 apps at once."
+            ),
+            title="Find, open & close apps on a Pixel phone",
+            # The task type for embedding. Check the available tasks in the model's documentation.
+            task_type="RETRIEVAL_DOCUMENT"
+        ),
+        # Option2: Use plain text as embedding input
```

Comment:
@msampathkumar , I'd really like to showcase the 2nd option here, which is using a plain text string as an input.

However, when I read this sample as an "external viewer", I immediately have a question:  "What will be the `task_type` of this input if I use a plain string?". And, unfortunately,  I can't answer this question myself.
I searched through the Vertex SDK source code, and I didn't find any hints for this 🤷 My guess is that `RETRIEVAL_DOCUMENT` is the default type. How can I prove or disprove this? Thanks!

Reply:
`model.get_embeddings`'s doc string has that the input has to be a list of texts or `TextEmbeddingInput` objects to embed. 

Function definition also says `List[Union[str, TextEmbeddingInput]]`. So I believe, simple text input is fine.